tracer/src/Datadog.Trace/Iast/ReturnedHeadersAnalyzer.cs

```diff
@@ -0,0 +1,160 @@
+        }
+        catch (Exception error)
+        {
+            Log.Error(error, $"{nameof(ReturnedHeadersAnalyzer)}.{nameof(Analyze)} exception");
+        }
+    }
+#else
+    // Analyze the headers. If the response is HTML, check for X-Content-Type-Options: nosniff. If it
+    // is not present, report a vulnerability. When getting the headers, make sure that keys are searched taking
+    // int account that can be case insensitive. Exclude vulnerability when return code is one of the ignorable.
+    internal static void Analyze(IHeaderDictionary responseHeaders, IntegrationId integrationId, string serviceName, int responseCode)
+    {
+        try
+        {
+            if (string.IsNullOrEmpty(serviceName) || IsIgnorableResponseCode((HttpStatusCode)responseCode))
+            {
+                return;
+            }
+
+            IastModule.OnExecutedSinkTelemetry(IastInstrumentedSinks.XContentTypeHeaderMissing);
+
+            string contentTypeValue = string.Empty;
+            string contentOptionValue = string.Empty;
+
+            // We iterate instead of trying to get the key directly because keys are case insensitive
+            foreach (var header in responseHeaders)
+            {
+                var headerKeyLow = header.Key.ToLowerInvariant();
+                if (headerKeyLow == ContentTypeLow)
+                {
+                    contentTypeValue = header.Value;
+
+                    if (!IsHtmlResponse(contentTypeValue))
+                    {
+                        return;
+                    }
+                }
+
+                if (headerKeyLow == XContentTypeOptionsLow)
+                {
+                    contentOptionValue = header.Value;
+
+                    if (IsNoSniffContentOptions(contentOptionValue))
+                    {
+                        return;
+                    }
+                }
+            }
+
+            LaunchXContentTypeOptionsVulnerability(integrationId, serviceName, contentTypeValue, contentOptionValue);
+        }
+        catch (Exception error)
+        {
+            Log.Error(error, $"{nameof(ReturnedHeadersAnalyzer)}.{nameof(Analyze)} exception");
+        }
+    }
+#endif
+
+    private static void LaunchXContentTypeOptionsVulnerability(IntegrationId integrationId, string serviceName, string contentTypeValue, string contentOptionValue)
+    {
+        if (!string.IsNullOrEmpty(contentTypeValue))
+        {
+            IastModule.OnXContentTypeOptionsHeaderMissing(integrationId, contentOptionValue, serviceName);
+        }
+    }
+
+    private static bool IsHtmlResponse(string contentTypeValue)
+    {
+        var contentType = contentTypeValue.ToLowerInvariant();
+        return contentType.Contains("text/html") || contentType.Contains("application/xhtml+xml");
+    }
+
+    private static bool IsNoSniffContentOptions(string contentOptionValue)
+    {
+        return contentOptionValue.ToLowerInvariant() == "nosniff";
```

Comment:
I think it's better to do this rather than allocating (applies basically everywhere) 😄

```suggestion
        return string.Equals(contentOptionValue, "nosniff", StringComparison.OrdinalIgnoreCase);
```